workers/generic-worker/fileutil/fileutil_posix.go

```diff
@@ -55,3 +56,26 @@ func SecureFiles(filepaths ...string) (err error) {
 	}
 	return nil
 }
+
+func ResetPermissions(path string, permissions fs.FileMode) error {
```

Comment:
Shouldn't this be SetPermissions? I know that we call it when we want to reset the permissions, but the function body itself takes a path and some permissions, and applies them, so really it is just setting the applied permissions, I would say.